### PR TITLE
OCPBUGS-83284: validate more CMO configmap fields

### PR DESCRIPTION
**Pull request by simonpasquier**
This commit moves some of the validations that were performed when reconciling the resources to the loading of the configuration files.

### CODE REVIEW

**Comment by simonpasquier** [comment]
@coderabbitai review

File: pkg/manifests/config.go [hunk 1]
```diff
@@ -343,27 +348,76 @@ func NewConfigFromString(content string) (*Config, error) {
...
+
+	if dirPath == "/" {
+		return errors.New("query log file can't be stored on the root directory")
+	}
+
+	if dirPath == "/dev" && path != "/dev/stdout" && path != "/dev/stderr" && path != "/dev/null" {
+		return errors.New("query log file can't be stored on a new file on the dev directory")
+	}
+
+	return nil
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/manifests/config.go:398]: We didn't bother about these edge cases in the previous implementation, the existing unit tests all pass so I wouldn't bother to complicate the validation (at least not in this PR).

**Comment by simonpasquier** [comment]
/hold

legit test failure

**Comment by simonpasquier** [comment]
/hold cancel

the test has been fixed

**Comment by machine424** [comment]
thanks for this.
How about defining custom types or/and have custom UnmarshalJSON for them? we do that for ExternalLabels https://github.com/openshift/cluster-monitoring-operator/blob/f574b53d9110a1a705e320d0e4f79c62d80b690a/pkg/manifests/config.go#L765 (the unmarshaller should give all needed context now in case of failures). So we don't have to do any wiring.

EDIT: and we'll be moving closer to the CRD based config.

**Comment by simonpasquier** [comment]
@machine424 not opposed to this but I wanted the refactoring to change as little things as possible.

**Comment by danielmellado** [comment]
Agree with Simon there, let's get this in first and later we can modify whatever, and CRDs are coming, yes xD

**Comment by machine424** [comment]
Apparently I never pushed some comments/approval. Never mind.
Fortunately, Dani is always around.

(As a reviewer I'd have accepted the UnmarshalJSON approach as both approaches would result in the same amount of changes (the logic is the same, only the wrapping changes); we've been moving those validate()/check() around many times, I think UnmarshalJSON would be a good place for them to live, for now...)
